src/Microsoft.ApplicationInsights/DataContracts/AvailabilityTelemetry.cs

```diff
@@ -178,6 +178,26 @@ public ITelemetry DeepClone()
             return new AvailabilityTelemetry(this);
         }
 
+        /// <summary>
+        /// Deeply clones the Extension of <see cref="AvailabilityTelemetry"/> object.
+        /// </summary>
+        IExtension IExtension.DeepClone()
+        {
+            return new AvailabilityTelemetry(this);
+        }
+
+        /// <inheritdoc/>
+        public void Serialize(ISerializationWriter serializationWriter)
+        {            
+            serializationWriter.WriteProperty("name", this.WriteTelemetryName(TelemetryName));
+            this.WriteEnvelopeProperties(serializationWriter);
+            Utils.CopyDictionary(this.Context.GlobalProperties, this.Data.properties);
```

Comment:
serialize shouldn't mutate an object. Move dictionary copying some other place

Reply:
Addressed this.